### PR TITLE
Make it a flake

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

I needed it to use as a flake and the current version would use impure `builtin.currentSystem` and `fetchGit`.

###### Things done

- [ ] Compiled all jobs with `nix build -f release.nix`
- [ ] Ran all tests with `nix build -f release.nix test`
- [x] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/thoughtpolice/eris/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,40 +1,33 @@
...
+, DBI
+, DBDSQLite
+, xz
+, bzip2
 }:
 
-with builtins;
-
-let
-  bootstrap = import ./nix/bootstrap.nix {
```
Comment (Mic92) [inline default.nix:13]: Sorry for this radical change. I needed to get something working quick and this was the easiest option.

Comment (Mic92) [reply default.nix:13]: In long term a lot of these git version information can be taken from the flake itself.

**Comment by Mic92** [comment]
I currently don't have the resources to look into change requests, but feel free to take it as an inspiration or as it is.

**Comment by thoughtpolice** [comment]
@Mic92 Oh, interesting, thanks. I really haven't kept up with Flakes in quite a long time, I admit, so I wasn't sure what the status of them are or standardize their usage in any of my own personal work.

I have a spare machine (rpi4) with Nix, so I'll look into if it's possible to add Flake support to my `bootstrap.nix` project easily when I get a chance, which would also make the remainder of my projects easily Flake compatible too, at least as easily as possible...

**Comment by Mic92** [comment]
> @Mic92 Oh, interesting, thanks. I really haven't kept up with Flakes in quite a long time, I admit, so I wasn't sure what the status of them are or standardize their usage in any of my own personal work.
> 
> I have a spare machine (rpi4) with Nix, so I'll look into if it's possible to add Flake support to my `bootstrap.nix` project easily when I get a chance, which would also make the remainder of my projects easily Flake compatible too, at least as easily as possible...

You can also just enable it for a different user on the same machine if that is any help (the user needs to be trusted)

**Pull request closed without merging**